.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,8 @@
+- id: formatr
+  name: formatR
+  description: Uses formatR to format R files
+  entry: scripts/pre-commit-formatR.R
+  language: script
+  types: [text]
```

Comment:
Not sure this has any effect if you specify `files` in the next line.

Reply:
It does. pre-commit's documentation specifies that it performs a logical AND on this and files. So it does restrict the files considered by the hook. Perhaps unnecessary but it can't do any harm.

Reply:
I don't ever use `types` in {precommit}'s [`.pre-commit-hooks.yaml`](https://github.com/lorenzwalthert/precommit/blob/main/.pre-commit-hooks.yaml), so I guess the logical and of an empty key key1 and a key key2 is just the key2. You are right, it does not harm.